### ISSUE TITLE
[Request]: Review APM getting started landing page

### ISSUE THREAD

**Issue by colleenmcginnis** [elastic/observability-docs#4336]
### Description

In https://github.com/elastic/observability-docs/pull/4230, @bmorelli25 brought up a few issues with the [APM getting started landing page](https://www.elastic.co/guide/en/observability/current/apm-getting-started-apm-server.html) where we compare/contrast the two approaches: [APM Server binary](https://www.elastic.co/guide/en/observability/current/apm-getting-started-apm-server.html#apm-setup-apm-server-binary) and [Fleet-managed APM Server](https://www.elastic.co/guide/en/observability/current/apm-getting-started-apm-server.html#apm-setup-fleet-managed-apm). We should review this whole page to make sure it's still accurate.

Specific feedback from https://github.com/elastic/observability-docs/pull/4230:

* At the top of the page we say _The easiest way to get started with Elastic APM is by using our hosted Elasticsearch Service on Elastic Cloud_, but in the APM Server binary's list of pros we say _Simplest self-managed option_.
  
  From @bmorelli25 via [comment](https://github.com/elastic/observability-docs/pull/4230#discussion_r1752867997):
  > At the top of the page we link to the fleet server method and say that's the easiest way to get started. Here we say the opposite.

* We have a note that says _Fleet-managed APM Server does not have full feature parity with the APM Server binary method of running Elastic APM._

  From @bmorelli25 via [comment](https://github.com/elastic/observability-docs/pull/4230#discussion_r1752875441):
  >Is this still the case? In looking through Slack, @simitt recently mentioned that other than fewer outputs, there is general feature parity. This might be stale.

* In the list of required components for the Fleet-managed option, we refer to APM Server, Elastic Agent, and Fleet Server individually.

  From @bmorelli25 via [comment](https://github.com/elastic/observability-docs/pull/4230#discussion_r1752877780):
  >I don't think this is fair. Elastic agent gives you Fleet Server and APM Server.

### Resources

Initial comments in https://github.com/elastic/observability-docs/pull/4230

### Which documentation set does this change impact?

Stateful only

### Feature differences

Stateful only

### What release is this request related to?

N/A

### Collaboration model

The documentation team

### Point of contact.

**Main contact:** @colleenmcginnis 

**Stakeholders:** @bmorelli25 @simitt

**Comment by simitt** [comment]
>At the top of the page we say The easiest way to get started with Elastic APM is by using our hosted Elasticsearch Service on Elastic Cloud, but in the APM Server binary's list of pros we say Simplest self-managed option.

Getting started with hosted version vs on-prem -> hosted version is simpler. While it is the Fleet managed version, we abstract away and handle all the initial setup and configuration for the users, so they have a much simpler time getting started. However, if chosen to run on-prem, then apm-server binary is much simpler to get started with than the Fleet managed version. 
TL;DR - this is still true, but maybe the wording could be improved

>We have a note that says Fleet-managed APM Server does not have full feature parity with the APM Server binary method of running Elastic APM.

The missing support for outputs different than Elasticsearch is the only thing that comes to mind ad-hoc, but I also haven't spent much time on this recently. We should still be calling out that we only support ES as output when running under Fleet. 

>In the list of required components for the Fleet-managed option, we refer to APM Server, Elastic Agent, and Fleet Server individually.

Fair; the standard user does not have to know this, but if there is any problem or troubleshooting it is good to know that Elastic Agent actually is running multiple sub processes.
